app/models/text.rb

```diff
@@ -26,6 +26,7 @@ class Text < ApplicationRecord
   has_many :read_texts, dependent: :destroy
   has_many :movies
   belongs_to :genre, optional: true
+  has_many :progresses, dependent: :destroy
```

Comment:
`as: :materiable` を入れ忘れていますよ。これを使えば `text.progresses` の書き方ができるので，無理な書き方をしなくてよくなるはずです。


```suggestion
  has_many :progresses, dependent: :destroy, as: :materiable
```